### PR TITLE
feature/VETHUB-63

### PR DESCRIPTION
**Pull request by freshc0w**
To setup, please git clone the repo and follow instructions in the README

**Testing:**
```bash
git fetch
git checkout feature/VETHUB-63
// make a folder 'logs' and make a csv folder named 'broken-links-logs.csv'
```
image

```bash
cd data
node validateQuestionLink.js
```

Take a look at the terminal for more information about the validation process.

### CODE REVIEW

**Comment by anzau** [comment]
Can we add this as part of `package.json` ?

File: data/validateQuestionLink.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    console.error(
+      `An error occured while accessing "${input}". \nError: ${error}`,
+    );
+    return false;
+  }
+};
+
+const validateQuizLink = async (linkList) => {
+  console.info("validating links...", linkList.length);
+  const badLinks = [];
```
Comment (anzau) [inline data/validateQuestionLink.js]: Maybe we can assign `badLinks` to the `Promise.all` and then retrieve the links in the return.

File: data/validateQuestionLink.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { questions } from "./question-data.js";
+import fs from "fs/promises";
+import path from "path";
+
+// Run this script and check logs/broken-links-logs.csv for broken links
+
+const STATUS_SUCCESS = 200;
```
Comment (anzau) [inline data/validateQuestionLink.js]: Let's keep the naming conventions consistent and use camelCase.

File: data/validateQuestionLink.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  );
+  return badLinks;
+};
+
+async function logError(errorLog, message) {
+  await fs.appendFile(errorLog, message + "\n");
+}
+
+/**
+ * Sometimes multiple retries in validation of links is required since there
```
Comment (anzau) [inline data/validateQuestionLink.js:55]: Are they all running asynchronously? Can we not make it run synchronously?

**Comment by freshc0w** [comment]
> Can we add this as part of `package.json` ?

Sorry, which part? The added `--port 5712`?

File: data/validateQuestionLink.js (hunk 1 above)
Comment (freshc0w) [reply data/validateQuestionLink.js]: fixed

File: data/validateQuestionLink.js (hunk 2 above)
Comment (freshc0w) [reply data/validateQuestionLink.js]: fixed

File: data/validateQuestionLink.js (hunk 3 above)
Comment (freshc0w) [reply data/validateQuestionLink.js:55]: Unsure about the question. The `badLinks` is dependent on the previous operation where the previous `badLinks` were checked so I think no? Feel free to give me an example so I can understand better.

Comment (anzau) [reply data/validateQuestionLink.js:55]: I was just wondering what was the reason behind running it few times (As in if there is a function running asynchronously where if you run it in a few attempts, it's all good).

**Comment by anzau** [comment]
> > Can we add this as part of `package.json` ?
> 
> Sorry, which part? The added `--port 5712`?

I forgot to add the reference sorry, this command: `node validateQuestionLink.js`.

**Review by Jo-Zh — APPROVED** [review]

**Pull request merged**